sidecar.sh

```diff
@@ -43,35 +42,56 @@ populate_notebook() {
     MANIFESTS=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/")
-    while [ -z "$MANIFESTS" ]; do
-        log "Unable to get manifests from '$GEN3_ENDPOINT/manifests/'"
-        log $MANIFESTS
+    log "querying manifest service at $GEN3_ENDPOINT/metadata/"
+    METADATA=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/metadata/")
+
+    while [ -z "$MANIFESTS" ] || [ -z "$METADATA" ]; do
+        if [ -z "$MANIFESTS" ]; then
+            log "Unable to get manifests from '$GEN3_ENDPOINT/manifests/'"
+            log $MANIFESTS
+        fi
+        if [ -z "$METADATA" ]; then
+            log "Unable to get metadata from '$GEN3_ENDPOINT/metadata/'"
+            log $METADATA
+        fi
         log "sleeping for 15 seconds before trying again.."
         sleep 15
         MANIFESTS=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/")
+        METADATA=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/metadata/")
     done
+    log "successfully retrieved manifests and metadata for user"
 
+    process_files() {
+        local data=$1
+        local base_dir=$2
 
-    #  Loop over each exported manifest
-    echo $MANIFESTS | jq -c '.manifests[]' | while read i; do
+        echo $data | jq -c '.[]' | while read i; do
         FILENAME=$(echo "${i}" | jq -r .filename)
         FOLDERNAME=$(echo "${FILENAME%.*}")
-        FOLDER="/data/${GEN3_ENDPOINT}/exported-${FOLDERNAME}"
+        FOLDER="/data/${GEN3_ENDPOINT}/exported-${base_dir}/exported-${FOLDERNAME}"
 
         if [ ! -d "$FOLDER" ]; then
             log "mkdir -p $FOLDER"
             mkdir -p $FOLDER
 
             # make sure folder can be written to by notebook
             chown -R 1000:100 $FOLDER
-            MANIFEST=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/file/$FILENAME")
-            echo "${MANIFEST}" > $FOLDER/manifest.json
 
-            log "Creating notebook for $FILENAME"
-            cp ./template_manifest.json $FOLDER/data.ipynb
-            populate_notebook "$MANIFEST" "$FOLDER"
+            if ["$base_dir" == "manifests"];then
+                MANIFEST=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/file/$FILENAME")
+                echo "${MANIFEST}" > $FOLDER/manifest.json
+                log "Creating notebook for $FILENAME"
+                cp ./template_manifest.json $FOLDER/data.ipynb
+                populate_notebook "$MANIFEST" "$FOLDER"
+            elif ["$base_dir" == "metadata"];then
+                METADATA=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/metadata/$FILENAME")
+                echo "${METADATA}" > $FOLDER/metadata.json
+            fi
         fi
-    done
+        done
+    }
+    echo $MANIFESTS | jq -c '.manifests' | process_files manifests
```

Comment:
I think this is not the correct way to pass arguments in bash: if you pipe the output, it will go to stdin, then I think you cannot read them the same way as command line arguments by `$`
So maybe something like this would work instead `process_files "manifests" "$(echo $MANIFESTS | jq -c '.manifests')"`, we need to test that

Reply:
Thanks Mingfei, that's a good point. I switched the order of parameters in `process_files()` and made the change to lines 93/94, commenting out the old code for now